### PR TITLE
Fix LLVM 5.0 build

### PR DESCRIPTION
**Pull request by JohanEngelen**

### CODE REVIEW

**Comment by thewilsonator** [comment]
I'll have a new LLVM up for the SPIRV OSX box soon.

**Comment by thewilsonator** [comment]
Done. Not sure how to force an de-chaching.

**Comment by JohanEngelen** [comment]
> Not sure how to force an de-chaching.

You could version the llvm-spirv dir (`llvm-spirv-5.0.0-abc123`).
You can also delete caches ("more options" top right), but then all the subbuilds will need to recache.

**Comment by thewilsonator** [comment]
Thanks Johan. Sorry I working like mad on my honours, I'll be out of action until 5th of July.

**Comment by JohanEngelen** [comment]
SPIRV build is green again.

File: driver/main.cpp [hunk 1]
```diff
@@ -118,33 +118,45 @@ static inline llvm::Optional<llvm::Reloc::Model> getRelocModel() {
...
 
-void printVersion() {
-  printf("LDC - the LLVM D compiler (%s):\n", global.ldc_version);
-  printf("  based on DMD %s and LLVM %s\n", global.version,
-         global.llvm_version);
-  printf("  built with %s\n", ldc::built_with_Dcompiler_version);
+// This function exits the program.
+void printVersion(llvm::raw_ostream &OS) {
+  OS << "LDC - the LLVM D compiler (" << global.ldc_version << "):\n";
+  OS << "  based on DMD " << global.version << "and LLVM " << global.llvm_version << "\n";
```
Comment (kinke) [inline driver/main.cpp]: Missing space before `and LLVM`.

**Comment by JohanEngelen** [comment]
:( travis jobs keep running out of time

**Comment by kinke** [comment]
Yeah on weekends the Travis performance is a catastrophe. :/

**Comment by JohanEngelen** [comment]
It looks like we really need to find an alternative to Travis, or speed up our builds by a lot :(
I'm tired of not being able to merge things because of Travis issues :/

**Comment by wilzbach** [comment]
> I'm tired of not being able to merge things because of Travis issues :/

Not that it would help with the timeouts on Travis, but I am happy to setup auto-merge with the Dlang-Bot in case that would help you. Currently it doesn't restart failing transient Travis builds yet, but that is on our TODO list (of course it will happen faster with a PR).

**Comment by kinke** [comment]
Rebased after merging the Travis tweaks in #2074. Travis time-outs should now be exceedingly rare; Circle a) relies on the flaky LLVM APT repo and b) currently exhibits a new/temporary (?) failure of `runnable/gdb4149.d`; let's hope the rest is finally green now again (without having to retrigger Travis jobs manually).

**Comment by thewilsonator** [comment]
Circle has had that GDB failure for quite some time (months), unless it got fixed and its a new failure.

**Pull request merged**